Dockerfile

```diff
@@ -4,43 +4,38 @@ ENV USER l4d2
 ENV HOME /home/$USER
 ENV SERVER $HOME/hlserver
 
+ENV DEBIAN_FRONTEND noninteractive
+ENV DEBCONF_NONINTERACTIVE_SEEN true
+
 RUN apt-get -y update \
-    && apt-get -y upgrade \
-    && apt-get -y install lib32gcc1 curl net-tools lib32stdc++6 locales unzip lib32z1 screen \
+    && apt-get --no-install-recommends -y install lib32gcc1 curl net-tools lib32stdc++6 locales unzip lib32z1 screen \
```

Comment:
Tested this out tonight. I got SSL errors on the steamcmd usage. Removing `--no-install-recommends` here fixes that issue.

Reply:
OK. So I need to figure out _which_ package in the list of recommends is actually required for us, and explicitly install it, I think.

Reply:
It's probably the `ca-certificates` package. But, why can't we just leave recommends in place? This is going to run as an internet-connected server; I think we should install stuff that the OS recommends.

Reply:
It's not a big deal either way in this case. What motivates my thinking on this is that it's very often a best practice to know exactly what you're putting in the image you're distributing: what is in "recommends" can change over time and the decisions about what to put in there aren't driven by what makes sense to have in docker. In more widely distributed images, or with something where low cold-start time is valuable, image size is also more of a concern than it probably is here. 

This is also part of why `COPY . .` (which we aren't doing here, thankfully) is widely regarded as an anti-pattern/smell: you don't really know what you're putting in the image you're distributing, but you can be sure you're putting in stuff that doesn't need to be there. So, I don't mind figuring out what's missing (and I think your guess is most likely correct). 

Having said all that, if you prefer to just leave the "recommends" in, I understand completely and we can go that route.

Reply:
Thanks for walking me through that. I think we should just leave in the recommends for now. A lot of the external sources could change from under us. I'd rather not miss out on security patches or CA cert revocations because we disabled this.

Reply:
f47ece151884feab27aa89fe2a2ff75a4d8b98b2